Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -883,3 +894,23 @@ internal func assertExpression(_ body: () -> Bool) {
         return body()
     }())
 }
+
+extension SelectableEventLoop {
+    @inlinable
+    func scheduleCallback(at deadline: NIODeadline, handler: some NIOScheduledCallbackHandler) -> NIOScheduledCallback {
+        let taskId = self.scheduledTaskCounter.loadThenWrappingIncrement(ordering: .relaxed)
+        let task = ScheduledTask(id: taskId, handler, deadline)
+        try! self._schedule0(.scheduled(task))
```

Comment:
Can you drop a comment here as to why the `try!` is safe?

Reply:
I think (2) is fine

Reply:
After some offline discussion it turns out that I was mistaken. Scheduling _can_ throw in other circumstances, not just when the EL is shutdown.

Made `scheduleCallback` functions throwing in 9bf65f610.